packages/fondue/src/components/Tooltip/Tooltip.tsx

```diff
@@ -36,22 +36,21 @@ const formatTooltipText = (text: string) => {
     return text.split(lineBreakRegex).join('\n');
 };
 
-type TimeoutRef = {
-    current: NodeJS.Timeout | null;
-};
-
-const handleTimeout = (callback: () => void, delay: number, timeoutRef: TimeoutRef) => {
-    if (timeoutRef.current) {
-        clearTimeout(timeoutRef.current);
+let timeoutId: NodeJS.Timeout | undefined;
```

Comment:
Please put it in the component, having it outside is a side-effect.

Reply:
I didn't put the function back I passed the ref as prop so that the function can leave outside of the main component. 
Great catch for the side effect, thx

Reply:
Updated